gateway-sp-comms/src/sp_response_expect.rs

```diff
@@ -348,6 +351,24 @@ mod tests {
             panic!("mismatched value {v:?}");
         };
         assert_eq!(r, VersionedSpState::V2(state));
+
+        let state = SpStateV3 {
+            hubris_archive_id: [1, 2, 3, 4, 5, 6, 7, 8],
+            serial_number: [0; 32],
+            model: [0; 32],
+            revision: 123,
+            base_mac_address: [0; 6],
+            power_state: gateway_messages::PowerState::A0,
+        };
+        let v = expect_sp_state((
+            dummy_addr(),
+            SpResponse::SpStateV3(state),
+            vec![],
+        ));
+        let Ok(r) = v else {
+            panic!("mismatched value {v:?}");
+        };
+        assert_eq!(r, VersionedSpState::V3(state));
     }
```

Comment:
I'm pretty sure if we're adding new entries here we need a protocol bump and corresponding tests (c.f. https://github.com/oxidecomputer/management-gateway-service/commit/e727e960d31f62afe6077fdfc405e0ac2379f6b2)

Reply:
I added tests to gateway_messages/tests/versioning/v12.rs that include SpStateV3 and other new symbols.